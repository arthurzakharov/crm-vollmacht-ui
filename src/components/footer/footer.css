.footer {
  display: flex;
  flex-direction: column;
}

.footer__line {
  display: none;
}

.footer__content {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
  font-family: "Open Sans", sans-serif;
  font-weight: 300;
  font-size: 14px;
  line-height: 1.5;
  color: var(--text-secondary);
  white-space: nowrap;
}

.footer__links {
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
  list-style: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.footer__link {
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
}

.footer__link:last-of-type .footer__separator {
  display: none;
}

.footer__separator {
  border-right: 1px solid var(--text-secondary);
  margin-right: 6px;
  margin-left: 6px;
  height: 16px;
}

@media screen and (min-width: 768px) {
  .footer__line {
    display: block;
    border: none;
    margin-bottom: 10px;
    height: 2px;
    background-color: var(--footer-line);
  }

  .footer__content {
    flex-direction: row;
    justify-content: space-between;
  }
}
