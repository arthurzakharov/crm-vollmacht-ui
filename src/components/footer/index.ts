export { default, FooterProps } from "./footer";
