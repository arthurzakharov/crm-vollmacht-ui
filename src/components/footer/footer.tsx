import React, { FC } from "react";
import "./footer.css";

export type FooterLink = {
  text: string;
  onClick: () => void;
};

interface Props {
  name: string;
  links: FooterLink[];
}

const Footer: FC<Props> = (props) => (
  <footer className="footer">
    <hr className="footer__line" />
    <div className="footer__content">
      <span>
        Â© {new Date().getFullYear()} {props.name}
      </span>
      <ul className="footer__links">
        {props.links.map((link: FooterLink) => (
          <li key={link.text} className="footer__link">
            <button type="button" tabIndex={0} onClick={link.onClick}>
              {link.text}
            </button>
            <div className="footer__separator" />
          </li>
        ))}
      </ul>
    </div>
  </footer>
);

Footer.displayName = 'Footer';

export default Footer;
