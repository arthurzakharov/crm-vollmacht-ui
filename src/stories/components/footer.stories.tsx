import { Meta, StoryObj } from "@storybook/react";
import { within, userEvent, fn, expect } from "@storybook/test";
import { Footer } from "@components/footer";
import MaxWidthDecorator from "@stories/decorators/max-width-decorator";

type Story = StoryObj<typeof Footer>;

const meta: Meta<typeof Footer> = {
  title: "COMPONENTS/Footer",
  component: Footer,
  parameters: {
    MaxWidthDecorator: 1100,
  },
  decorators: [MaxWidthDecorator],
};

export const WithLinks: Story = {
  name: "With links",
  args: {
    name: "legal-one",
    links: [
      {
        text: "Allgemeine Informationen",
        onClick: fn(),
      },
      {
        text: "Datenschutz",
        onClick: fn(),
      },
      {
        text: "Impressum",
        onClick: fn(),
      },
    ],
  },
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);
    await step("Clicking links one by one, check if CBs are fired", async () => {
      if (args.links) {
        for (const link of args.links) {
          await userEvent.click(canvas.getByText(link.text));
          await expect(link.onClick).toHaveBeenNthCalledWith(1);
        }
      }
    });
  },
};

export const WithEmptyLinks: Story = {
  name: "With Empty links",
  args: {
    name: "legal-one",
    links: [],
  },
};

export const DefaultValues: Story = {
  name: "Default Values",
  args: {
    name: "legal-one",
    links: undefined,
  },
};

export default meta;
