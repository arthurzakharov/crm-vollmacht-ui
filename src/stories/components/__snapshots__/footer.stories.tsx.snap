// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS/Footer DefaultValues smoke-test 1`] = `
<footer class="footer">
  <hr class="footer__line">
  <div class="footer__content">
    <span>
      © 2024 legal-one
    </span>
    <ul class="footer__links">
    </ul>
  </div>
</footer>
`;

exports[`COMPONENTS/Footer WithEmptyLinks smoke-test 1`] = `
<footer class="footer">
  <hr class="footer__line">
  <div class="footer__content">
    <span>
      © 2024 legal-one
    </span>
    <ul class="footer__links">
    </ul>
  </div>
</footer>
`;

exports[`COMPONENTS/Footer WithLinks play-test 1`] = `
<footer class="footer">
  <hr class="footer__line">
  <div class="footer__content">
    <span>
      © 2024 legal-one
    </span>
    <ul class="footer__links">
      <li class="footer__link">
        <button type="button"
                tabindex="0"
        >
          Allgemeine Informationen
        </button>
        <div class="footer__separator">
        </div>
      </li>
      <li class="footer__link">
        <button type="button"
                tabindex="0"
        >
          Datenschutz
        </button>
        <div class="footer__separator">
        </div>
      </li>
      <li class="footer__link">
        <button type="button"
                tabindex="0"
        >
          Impressum
        </button>
        <div class="footer__separator">
        </div>
      </li>
    </ul>
  </div>
</footer>
`;
